File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> list = new ArrayList<>();
    public int getMinimumDifference(TreeNode root) {
        inorder(root);
        int min = Integer.MAX_VALUE;
        for(int i = 0; i < list.size()-1; i++){
            min = (list.get(i+1) - list.get(i)) < min ? (list.get(i+1) - list.get(i)) : min;
        }
        return min;
    }
    public void inorder(TreeNode root){
        if(root.left != null){
            inorder(root.left);
        }
        list.add(root.val);
        if(root.right != null){
            inorder(root.right);
        }
    }
}

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        queue = deque([root])
        lst = []
        while queue:
            size = len(queue)
            for i in range(size):
                node = queue.popleft()
                lst.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

        lst.sort()
        min = float('inf')
        for i in range(1,len(lst)):
            temp = lst[i] - lst[i-1] 
            if temp < min:
                min = temp
        return min

            
        


